docs/node-operators/how-to/rotate-ethereum-keys.md

```diff
@@ -8,104 +8,136 @@ hide_title: false
-* **ETH_NEW_ADDR** - Newly generated Ethereum address to rotate to
+* **ETH_ADDR** - The Ethereum address being rotated from
+* **ETH_NEW_ADDR** - The Ethereum address being rotated to
 * **TARGET_BLOCK** - The block in which the Ethereum key rotation will take place
-* **WALLET_NAME** - Name of wallet that to submit the transaction with
-* **SUBMITTER_ADDR** - Ethereum address of the submitter of produced signatures for [MultisigControl bridge](./../../api/bridge/interfaces/IMultisigControl). This needs to be an Ethereum account with some Ethereum to pay gas
-* **NODE_RPC_ADDR** - RPC address of any validator node in network (example: host-address:3002)
-* **NODE_REST_ADDR** - REST API address of any validator node in network (example: host-address:3003)
-* **DATA_NODE_REST_ADDR** - REST API address of any data node (example: host-address:3009)
-* **CLEF_ADDR** - Address of custom running Clef instance. Required only if Clef wallet is used (example: host-address:8550)
+* **SUBMITTER_ADDR** - An Ethereum address that will submit signatures to the [MultisigControl Bridge](./../../api/bridge/interfaces/IMultisigControl)
+* **DATA_NODE_REST_ADDR** - The REST API address of any data node
+* **CLEF_ADDR** - If rotating to a Clef Wallet, the address of the custom Clef instance
 
-## 1. Generate a new key - either by using node wallet or Clef
+## Generate a new Ethereum key
 
-### Using regular local store from node wallet
+Vega supports both local key-store wallets and [Clef](https://geth.ethereum.org/docs/clef/introduction) Wallets. Ethereum key rotation can be performed between wallets of any type. For example you can rotate from an key in a local key-store wallet, to a key generated in a Clef wallet.
+
+
+### Using the nodewallet local key-store
+The below command will generate a new Ethereum key in the local key-store:
 
 ```
 vega nodewallet generate --chain ethereum --force
 ```
 
-The command will output the response below, where the last bit of the file `01236c6afa74d90817d939682215907484cacd05` is the new account address.
-
-To set the **ETH_NEW_ADDR**, take the generated address and prefix it with 0x like so: `0x01236c6afa74d90817d939682215907484cacd05`.
-
+It will prompt for both the nodewallet's passphrase and the blockchain wallet's passphrase. The output from this command will look as follows:
 ```
 walletFilePath:
-path-to-wallet/UTC--2022-09-15T16-44-18.243348000Z--01236c6afa74d90817d939682215907484cacd05
+VEGA_HOME/node/wallets/ethereum/UTC--2022-09-30T13-38-37.768971000Z--0a2f0f6d4c1daad69193b5816aa6c3ba2ea5d483
+registryFilePath:
+VEGA_HOME/node/wallets.encrypted
 ```
 
-### Using Clef
-First, generate new account in Clef. This step will produce a new account address, referred to as **ETH_NEW_ADDR**. The newly generated account address will be printed to `stdout`.
+**ETH_NEW_ADDR** can be inferred from the `walletFilePath`. 
+
+For example the above case the new Ethereum address will be `0x0a2f0f6d4c1daad69193b5816aa6c3ba2ea5d483`
```

Comment:
Do you think it will be obvious enough to most users that they need to add 0x to the address above?

Reply:
I thought it was pretty crap anyway that the only way to get the address is to scrape some text off the end of some file. So now you've mentioned similar I might just change the import/generate commands to also print out the address with the `0x` and checksummed so that we can make it as simple aas possible.